app/src/main/java/org/mifos/mobile/viewModels/RegistrationViewModel.kt

```diff
@@ -0,0 +1,66 @@
+package org.mifos.mobile.viewModels
+
+import androidx.core.util.PatternsCompat
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.observers.DisposableObserver
+import io.reactivex.schedulers.Schedulers
+import okhttp3.ResponseBody
+import org.mifos.mobile.repositories.UserAuthRepositoryImp
+import org.mifos.mobile.utils.RegistrationUiState
+import javax.inject.Inject
+
+class RegistrationViewModel @Inject constructor(private var userAuthRepositoryImp: UserAuthRepositoryImp?) : ViewModel() {
```

Comment:
It should be UserAuthRepository, not the UserAuthRepositoryImp.

Reply:
Done

Reply:
I was expecting something like this 🤦 :
```
class RegistrationViewModel @Inject constructor(
     val userAuthRepositoryImp: UserAuthRepository
) : ViewModel() {
}
```

Reply:
my bad, fixed.